metricbeat/module/docker/diskio/diskio_test.go

```diff
@@ -1,14 +1,132 @@
 package diskio
 
 import (
+	"encoding/json"
 	"testing"
 	"time"
+
+	"github.com/elastic/beats/metricbeat/module/docker"
+
+	dc "github.com/fsouza/go-dockerclient"
 )
 
 var blkioService BLkioService
 var oldBlkioRaw = make([]BlkioRaw, 3)
 var newBlkioRaw = make([]BlkioRaw, 3)
 
+func TestDeltaMultipleContainers(t *testing.T) {
+	var apiContainer1 docker.Stat
+	var apiContainer2 docker.Stat
+	metrics := dc.BlkioStatsEntry{
+		Major: 123,
+		Minor: 123,
+		Op:    "Total",
+		Value: 123,
+	}
+	jsonContainers := `[
+     {
+             "Id": "8dfafdbc3a40",
+			 "Names": ["container"]
+     },{
+             "Id": "8dfafdbc3a41",
+			 "Names": ["container1"]
+     }]`
+	var containers []dc.APIContainers
+	err := json.Unmarshal([]byte(jsonContainers), &containers)
+	if err != nil {
+		t.Fatal(err)
+	}
+
+	apiContainer1.Stats.Read = time.Now()
+	apiContainer1.Container = containers[0]
+	apiContainer1.Stats.BlkioStats.IOServicedRecursive = append(apiContainer1.Stats.BlkioStats.IOServicedRecursive, metrics)
+	apiContainer2.Stats.Read = time.Now()
+	apiContainer2.Container = containers[1]
+	apiContainer2.Stats.BlkioStats.IOServicedRecursive = append(apiContainer2.Stats.BlkioStats.IOServicedRecursive, metrics)
+	dockerStats := make([]docker.Stat, 0)
+	dockerStats = append(dockerStats, apiContainer1)
+	dockerStats = append(dockerStats, apiContainer2)
+	stats := blkioService.getBlkioStatsList(dockerStats)
+	totals := make([]float64, 2, 2)
```

Comment:
No need to specify a capacity because the capacity of the slice is equal to its length if unspecified.